File: Leet 1461/solution.cs
public class Solution {
    public bool HasAllCodes(string s, int k) {
        int maxSize = 1 << k;
        HashSet<string> seen = [];
        
        for(int i = 0; i <= s.Length - k; i++)
        {
			seen.Add(s.Substring(i, k));

			if (seen.Count == maxSize)
            {
                return true;
            }
        }

		return false;
    }
}

File: Leet 1461/solution.py
class Solution:
    def hasAllCodes(self, s: str, k: int) -> bool:
        max_size: int = pow(2, k)
        
        seen: set[str] = set()
        
        for i in range(len(s) - k + 1):
            seen.add(s[i:i+k])

            if len(seen) == max_size:
                return True

        return False